engine/runtime/src/main/java/org/enso/interpreter/runtime/data/Type.java

```diff
@@ -519,29 +519,71 @@ private boolean isNothing(Node lib) {
   private Map<String, Function> methods() {
     if (methods == null) {
       CompilerDirectives.transferToInterpreter();
-      var allMethods = new HashMap<String, Function>();
-      var defScope = definitionScope.asModuleScope();
-      var methodsFromThisScope = defScope.getMethodsForType(this);
-      if (methodsFromThisScope != null) {
-        methodsFromThisScope.forEach(
+      methods = getMethods(true);
+    }
+    return methods;
+  }
+
+  /**
+   * Returns methods (both instance and static) defined on this type, including the ones inherited
+   * from Any. Instance methods are defined on this type, static methods are defined on its {@link
+   * #getEigentype() eigen type}. The methods defined on this type are searched for inside the
+   * module scope where this type is defined, so if there are any other extension methods defined in
+   * other modules, they are not included in the result.
+   *
+   * @param includeStaticMethods If static methods, defined on eigen type, should be included in the
+   *     result.
+   * @return All static and instance methods defined on this type, including the ones inherited from
+   *     Any.
+   */
+  @TruffleBoundary
+  public Map<String, Function> getMethods(boolean includeStaticMethods) {
+    var ctx = EnsoContext.get(null);
+    var allMethods = new HashMap<>(methodsFromAny(ctx));
```

Comment:
There is method `allTypes()` which returns _"All types this type represents including super types."_ - I guess it should be used here somehow to handle the _"numbers hierarchy"_.

Reply:
Fixed in 106b35d576f07f887428b3df695bfdf202a2f635